repos/system_upgrade/el7toel8/actors/multipathconfupdate/libraries/library.py

```diff
@@ -0,0 +1,249 @@
+
+_bool_options = {"retain_attached_hw_handler": True, "detect_prio": True,
+                 "detect_path_checker": True, "reassign_maps": False}
+
+_exist_options = ("hw_str_match", "ignore_new_boot_devs",
+                  "new_bindings_in_boot", "unpriv_sgio")
+
+_ovr_options = ("hardware_handler", "pg_timeout")
+
+
+class _QueueIfNoPathInfo(object):
+    def __init__(self, line, value):
+        self.line = line
+        self.value = value
+        self.has_no_path_retry = False
+
+
+def _nothing_to_do(config):
+    # pylint: disable=too-many-boolean-expressions,too-many-return-statements
+    if config.default_path_checker and config.default_path_checker != "tur":
+        return False
+    if config.default_retain_hwhandler is False or \
+            config.default_detect_prio is False or \
+            config.default_detect_checker is False or \
+            config.reassign_maps is True:
+        return False
+    if config.hw_str_match_exists or \
+            config.ignore_new_boot_devs_exists or \
+            config.new_bindings_in_boot_exists or \
+            config.unpriv_sgio_exists or \
+            config.detect_path_checker_exists or \
+            config.overrides_hwhandler_exists or \
+            config.overrides_pg_timeout_exists or \
+            config.queue_if_no_path_exists or \
+            config.all_devs_section_exists:
+        return False
+    if config.all_devs_options != []:
+        return False
+    return True
+
+
+def _comment_out_line(line):
+    return "# " + line + " # Commented out by Leapp"
+
+
+def _comment_out_ranges(lines, ranges):
+    for start, end in ranges:
+        line = lines[start]
+        lines[start] = "# " + line + " # Section commented out by Leapp"
+        for i in range(start + 1, end):
+            line = lines[i]
+            if line == "":
+                lines[i] = "#"
+            elif line[0] != '#':
+                lines[i] = "# " + line
+
+
+def _setup_value(value):
+    if re.search(r'\s', value):
+        return '"' + value + '"'
+    return value
+
+
+def _add_overrides(lines, options):
+    lines.append("overrides { # Section added by Leapp")
+    for option in options:
+        lines.append("\t{} {}".format(option.name, _setup_value(option.value)))
+    lines.append("}")
+    lines.append("")
+
+
+def _update_overrides(lines, ovr_line, options):
+    new_lines = []
+    start = None
+    for i, line in enumerate(lines):
+        if line is ovr_line:
+            start = i + 1
+            break
+    if start is None:
+        return
+    for option in options:
+        new_lines.append("\t{} {} # Line added by Leapp".
+                         format(option.name, _setup_value(option.value)))
+    lines[start:start] = new_lines      # insert new_lines
+
+
+def _convert_checker_line(line):
+    return line.replace("detect_path_checker", "detect_checker") + \
+            " # Line modified by Leapp"
+
+
+def _modify_features_line(line, value):
+    items = value.split()
```

Comment:
Isn't this a duplication of the qinp removal in the other actor?

Reply:
The difference is that the MultipathCornfRead actor is getting info that will be used to create a while new section, so it doesn't want to keep the existing line data, like indentation or comments.  While in MultipathConfUpdate, it is modifying a line in-place, and wants to keep all that information.